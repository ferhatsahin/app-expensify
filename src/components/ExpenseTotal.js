import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import numeral from 'numeral';
import selectExpenses from '../selectors/expenses';

export const ExpenseTotal = ({ expenses }) => {
  const expenseCount = expenses.length;
  const expenseWord = expenseCount > 1 ? 'expenses' : 'expense';
  return (
    <div className='page-header'>
      <div className='content-container'>
        <h1 className='page-header__title'>
          Viewing <span>{expenseCount}</span> {expenseWord}. Totaling{' '}
          <span>
            {numeral(expenses.map(expense => expense.amount).reduce((sum, value) => sum + value, 0) / 100).format(
              '$0,0.00'
            )}
          </span>
        </h1>
        <div className='page-header__actions'>
          <Link to='create' className='button'>
            Add Expense
          </Link>
        </div>
      </div>
    </div>
  );
};

const mapStatoToProps = state => ({
  expenses: selectExpenses(state.expenses, state.filters)
});

export default connect(mapStatoToProps)(ExpenseTotal);
