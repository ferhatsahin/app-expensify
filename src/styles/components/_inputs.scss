.text-input {
  border: 1px solid #dbdbdb;
  height: 48px;
  font-size: $font-size-large;
  font-weight: 300;
  padding: $s-size;
}

.select {
  @extend .text-input;
}

.text-area {
  @extend .text-input;
  height: 10rem;
}
