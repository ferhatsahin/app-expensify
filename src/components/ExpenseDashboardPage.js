import React from 'react';
import ExpenseList from './ExpenseList';
import ExpenseListFilters from './ExpenseListFilters';
import ExpenseTotal from './ExpenseTotal';

const ExpenseDashboardPage = () => (
  <div>
    <ExpenseTotal />
    <ExpenseListFilters />
    <ExpenseList />
  </div>
);

export default ExpenseDashboardPage;
