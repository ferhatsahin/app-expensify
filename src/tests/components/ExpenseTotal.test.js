import React from 'react';
import { shallow } from 'enzyme';
import { ExpenseTotal } from '../../components/ExpenseTotal';

test('Should render ExpenseTotal correctly', () => {
  const demoExpenses = [
    {
      description: 'rent',
      amount: 10950
    },
    {
      description: 'coffee',
      amount: 195
    }
  ];
  const wrapper = shallow(<ExpenseTotal expenses={demoExpenses} />);
  expect(wrapper).toMatchSnapshot();
});
