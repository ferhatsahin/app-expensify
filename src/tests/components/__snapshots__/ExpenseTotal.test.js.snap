// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render ExpenseTotal correctly 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        2
      </span>
       
      expenses
      . Totaling
       
      <span>
        $111.45
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        to="create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;
